web/packages/design/src/Modal/Modal.tsx

```diff
@@ -48,7 +48,7 @@ export type ModalProps = {
    * setAttribute on the outermost element. This is no longer the case, so technically this type can
    * be adjusted if needed.
    */
-  children?: React.ReactElement;
+  children?: React.ReactElement<any>;
```

Comment:
`React.ReactElement` seems to work just fine and in a number of other cases I checked.

https://react.dev/blog/2024/04/25/react-19-upgrade-guide#changes-to-the-reactelement-typescript-type

The way I understand it is that it's not needed in most cases. I think we should revert this. Unfortunately, this change is bundled with the codemod to `useRef`.

Reply:
i reverted the ones that didnt cause issues without .looks like mostly the `MenuAction` components require it

Reply:
I reverted a few more of them, should be fine now!